File: jvm-client/src/test/java/com/tomaszezula/makker/client/jvm/java/SetModuleData.java
package com.tomaszezula.makker.client.jvm.java;

import com.tomaszezula.makker.client.jvm.java.model.ModuleUpdate;

import java.util.Arrays;

import static com.tomaszezula.makker.client.jvm.java.Config.makeClient;

public class SetModuleData {
    public static void main(String[] args) {
        singleModule();
        multipleModules();
    }

    private static void singleModule() {
        makeClient.setModuleData(
                        471310,
                        9,
                        "value",
                        "{{5.greeting}}"
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }

    private static void multipleModules() {
        makeClient.setModuleData(
                        471310,
                        Arrays.asList(
                                new ModuleUpdate(9, "value", "{{5.greeting}}"),
                                new ModuleUpdate(12, "value", "{{5.greeting}}"),
                                new ModuleUpdate(13, "json", "{{5.greeting}}")
                        )
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }
}

File: jvm-client/src/test/kotlin/com/tomaszezula/makker/client/jvm/examples/SetModuleData.kt
package com.tomaszezula.makker.client.jvm.examples

import io.kotest.common.runBlocking
import com.tomaszezula.makker.client.jvm.examples.Config.makeClient
import com.tomaszezula.makker.client.jvm.model.ModuleUpdate
import com.tomaszezula.makker.common.model.Blueprint
import com.tomaszezula.makker.common.model.Scenario

fun main() {
    runBlocking {
        singleModule()
        multipleModules()
    }
}

private suspend fun singleModule() {
    makeClient.setModuleData(
        Scenario.Id(471310),
        Blueprint.Module.Id(9),
        "value",
        "{{5.greeting}}"
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}

private suspend fun multipleModules() {
    makeClient.setModuleData(
        Scenario.Id(471310),
        listOf(
            ModuleUpdate(Blueprint.Module.Id(9), "value", "{{5.greeting}}"),
            ModuleUpdate(Blueprint.Module.Id(12), "value", "{{5.greeting}}"),
            ModuleUpdate(Blueprint.Module.Id(13), "json", "{{5.greeting}}")
        )
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}
